stan/math/rev/scal/fun/rising_factorial.hpp

```diff
@@ -8,28 +8,23 @@
 namespace stan {
   namespace math {
 
-    namespace {
+    /**
+     * Return tangent of the rising factorial function 
+     * applied to the inputs. 
+     * Will throw for NaN x and for negative n, as 
+     * implemented in primitive function.
+     *
+     * @tparam T Scalar type of autodiff variable.
+     * @param x Argument.
+     * @param n Argument
+     * @return tangent of rising factorial at arguments.
+     */
 
-      class rising_factorial_vv_vari : public op_vv_vari {
-      public:
-        rising_factorial_vv_vari(vari* avi, vari* bvi) :
-          op_vv_vari(rising_factorial(avi->val_, bvi->val_),
-                     avi, bvi) {
-        }
-        void chain() {
-          avi_->adj_ += adj_
-            * rising_factorial(avi_->val_, bvi_->val_)
-            * (digamma(avi_->val_ + bvi_->val_)
-               - digamma(avi_->val_));
-          bvi_->adj_ += adj_
-            * rising_factorial(avi_->val_, bvi_->val_)
-            * digamma(bvi_->val_ + avi_->val_);
-        }
-      };
+    namespace {
```

Comment:
We want to get rid of these anonymous `namespace` with `namespace internal { ... }` and then everything in the namespace will need to be qualified with `internal::`.  The use of `internal` signifies that it's not something an API client is going to be able to rely on.